asdfghjkl/precondition.py

```diff
@@ -83,10 +87,25 @@ def accumulate_curvature(self, postfix='acc', scale=1., to_pre_inv=False, replac
                 continue
             fisher.scaling(scale)
             fisher_acc = self._get_fisher(module, postfix)
-            if fisher_acc is None or replace:
+            if fisher_acc is None:
                 self._set_fisher(module, fisher, postfix)
+            elif ema_decay == 1.0 or replace:
+                # We copy fields one by one because 'fisher_acc = fisher' causes
+                # losing inverse matrices
+                if fisher_acc.has_data:
+                    fisher_acc.data = fisher.data
+                if fisher_acc.has_kron:
+                    fisher_acc.kron.A = fisher.kron.A
+                    fisher_acc.kron.B = fisher.kron.B
+                if fisher_acc.has_diag:
+                    fisher_acc.diag.weight = fisher.diag.weight
+                    fisher_acc.diag.bias = fisher.diag.bias
+                if fisher_acc.has_unit:
+                    fisher_acc.unit.data = fisher.unit.data
```

Comment:
the `_pre_inv_postfix`, `_pre_inv_attr`, and `_get_pre_inv_fisher()` are prepared for avoiding this issue.
By explicitly specifying the `_pre_inv_postfix`, the fisher to store the inverses will be separated by that for the accumulation. So I don't see any issue with `fisher_acc = fisher` or `fisher_acc + fisher`

I know this is very confusing and we need an improved data structure for fisher.
we can discuss this at another issue or PR

Reply:
If you agree with avoiding the "losing inverse issue" by using `pre_inv_postfix,` can you remove your changes here? I'm open to discussing this point in more detail!

Reply:
I agree with there will be two fisher matrices if we enable `pre_inv_postfix` and `to_pre_inv=True`, and this is exactly what my script does. 

However, If we do `accumulate_curvature` and `update_inv` in different periods (intervals),  then we lose the inverse. For example, I do `accumulate_curvature` every step and do `update_inv` every 10 steps. 
The first step will be ok because we have no 'fisher_acc' so we just set it up.
At the second step we also do `accumulate_curvature`, then `fisher_acc = fisher` or `fisher_acc + fisher` will be called. And the inverse is gone. And we can't call `precondition`.

I think the biggest problem is that `accumulate_curvature` and `update_inv` share the same fisher matrix. And `accumulate_curvature` will overwrite it without conserving the inverse. If we don't `update_inv` in the same period, the `precondition` will fail.

So if we have two fisher matrices for `accumulate_curvature` and `update_inv` separately, the problem will be solved.

Reply:
Yes, as mentioned in #7, using `pre_inv_postfix` isn't a solution. 
And yes, having "two fisher matrices for accumulate_curvature and update_inv separately" is a solution.
But as mentioned in #7, this requires redundant memory consumption or makes the procedure a bit complicated.
So I think supporting the accumulating and inverting fisher in `SymMatrix` class is a simple solution.

That said, this modification is too much to discuss in this PR.
So my suggestion is to keep this part as it was before, merge this PR, and modify `SymMatrix` class in another PR which is correspond to #7